.github/workflows/ci.yml

```diff
@@ -74,7 +74,7 @@ jobs:
             path: html
 
   linux-testing:
-    name: ${{ matrix.os }}_${{ matrix.cxx_compiler }}_backend=${{ matrix.backend }}_device=${{ matrix.device_type }}_cxx${{ matrix.std }}_${{ matrix.build_type }}
+    name: ${{ matrix.device_type }},bknd=${{ matrix.backend }},cmplr=${{ matrix.cxx_compiler }},${{ matrix.os }},std${{ matrix.std }},cfg=${{ matrix.build_type }}
```

Comment:
If the overall length is not a limit, I suggest to change to `std=c++${{ matrix.std }}` (e.g. `std=c++17`) so that it is similar to the well-known compiler option.

Otherwise, the improvements look very good to me.